docs/composition.md

```diff
@@ -0,0 +1,882 @@
+---
+title: Publishing Infrastructure
+toc: true
+weight: 400
+indent: true
+---
+
+# Publishing Infrastructure
+
+Crossplane allows infrastructure operators to define new kinds of infrastructure
+resources and publish them for the application operators they support to use,
+all without writing any code.
+
+Infrastructure providers enable Crossplane to manage a wide array of
+infrastructure resources like Azure SQL servers and AWS ElastiCache clusters.
+Infrastructure composition allows infrastructure operators to define, share, and
+reuse new kinds of infrastructure resources that are _composed_ of other
+infrastructure resources. Infrastructure operators may configure one or more
+compositions of any defined resource, and may publish any defined resource to
+their application operators, who may then declare that their application
+requires that kind of resource.
+
+Composition can be used to build a catalogue of kinds and configuration classes
+of infrastructure that fit the needs and opinions of your organisation. As an
+infrastructure operator you might define your own `MySQLInstance` resource. This
+resource would allow your application operators to configure only the settings
+that _your_ organisation needs - perhaps engine version and storage size. All
+other settings are deferred to a selectable composition representing a
+configuration class like "production" or "staging".
+
+> Note that composition is an **experimental** feature of Crossplane. Refer to
+> [Current Limitations] for information on functionality that is planned but not
+> yet implemented.
+
+## Concepts
+
+![Infrastructure Composition Concepts]
+
+A _Composite_ infrastructure resource is composed of other resources. Its
+configuration schema is user-defined. The `MySQLInstance` resources in the above
+diagram are composite infrastructure resources.
+
+A `Composition` specifies a set of resources of which a composite infrastructure
+resource may be composed. The Azure and AWS `Composition` resources in the above
+diagram configure two possible compositions of a `MySQLInstance`.
+
+A _Requirement_ for an infrastructure resource declares that an application
+requires a particular kind and configuration of composite infrastructure
+resource. The `MySQLInstanceRequirement` resources in the above diagram declare
+that the application pods each require a `MySQLInstance`.
+
+An `InfrastructureDefinition` defines a new kind of composite infrastructure
```

Comment:
Could we use the term `InfrastructureDefinition` in place of "composite infrastructure resource" in this document? Could make the connections more clear and give us proper nouns to build understanding around.
"A _Composite_ infrastructure resource is composed of other resources. Its
configuration schema is user-defined" by an `InfrastructureDefinition`. In the diagram above, `MySQLInstance` has been defined by the `MySQLInstance` `InfrastructureDefinition`."

Reply:
In short, I'd rather not. In this case the thing and the definition of the thing are distinct concepts that I'd rather avoid conflating. I do take your point that "composite infrastructure resource" is a mouthful and that something more pithy would be ideal, though.